coordinator/src/main/java/com/tencent/rss/coordinator/SimpleClusterManager.java

```diff
@@ -72,14 +72,18 @@ public SimpleClusterManager(CoordinatorConf conf) {
     }
   }
 
-  private void nodesCheck() {
+  @VisibleForTesting
+  protected void nodesCheck() {
     try {
       long timestamp = System.currentTimeMillis();
       Set<String> deleteIds = Sets.newHashSet();
       for (ServerNode sn : servers.values()) {
         if (timestamp - sn.getTimestamp() > heartbeatTimeout) {
           LOG.warn("Heartbeat timeout detect, " + sn + " will be removed from node list.");
           deleteIds.add(sn.getId());
+          for (Set<ServerNode> nodesWithTag : tagToNodes.values()) {
+            nodesWithTag.remove(sn);
+          }
         }
       }
       for (String serverId : deleteIds) {
```

Comment:
How about processing the delete nodesWithTag logic here? And cc @colinma

Reply:
Sorry, it's my fault. Put together with `deleteIds`'s logic will be better.